client.go

```diff
@@ -10,40 +13,165 @@ import (
 
 type Client struct {
 	endpoint string
-	apikey   string
+	auth     APIAuth
 	Limit    int
 	Offset   int
 	*http.Client
 }
 
 const NoSetting = -1
+const (
+	AuthTypeBasicAuth = iota
+	AuthTypeTokenQueryParam
+	AuthTypeBasicAuthWithTokenPassword
+	AuthTypeNoAuth
+)
+
+const (
+	httpMethodGet    = "GET"
+	httpMethodPost   = "POST"
+	httpMethodPut    = "PUT"
+	httpMethodDelete = "DELETE"
```

Comment:
Why are you defining new constants for the http methods? You can use predefined constants from the `http` package e. g. `http.MethodPut`.

Reply:
Done with commit 939fb234f5541b20bacbb56ea6bdb9efcd75b29d